test/framework/src/main/java/org/elasticsearch/test/RandomObjects.java

```diff
@@ -185,14 +185,21 @@ public static BytesReference randomSource(Random random, XContentType xContentTy
 
     /**
      * Randomly adds fields, objects, or arrays to the provided builder. The maximum depth is 5.
+     * @return Returns the number of fields in the document
      */
-    private static void addFields(Random random, XContentBuilder builder, int minNumFields, int currentDepth) throws IOException {
-        int numFields = randomIntBetween(random, minNumFields, 10);
+    private static int addFields(Random random, XContentBuilder builder, int minNumFields, int currentDepth,
+                                 int currentFields) throws IOException {
+        int maxTotalFields = 200;
```

Comment:
I'm not sure what value is good for this. From my results, 200 seemed to give decently big documents without going beyond 100KB.

Reply:
I don't think that we need such big docs, 100kb seems a lot still even if it's the upper bound.